app/models/ticket_log.rb

```diff
@@ -13,6 +14,17 @@ class TicketLog < ApplicationRecord
     include: %w[ticket_log_changes],
   }.freeze
 
+  validate :validate_metadata_action
+  def validate_metadata_action
+    errors.add(:metadata_action, "can only be set when action_type is metadata_action") if action_type != TicketLog.action_types[:metadata_action] && metadata_action.present?
```

Comment:
You can use `if !self.action_type_metadata_action?` which is autogenerated by Rails thanks to your `prefix: true` at the enum declaration